packages/styles/src/components/form-check.scss

```diff
@@ -206,61 +210,162 @@
+      background-color: tokens.get('switch', 'handle-enabled');
+      border-radius: tokens.get('switch', 'border-radius-round');
+      transition:
+        transform animation.$transition-base-timing,
+        background-color animation.$transition-base-timing;
+    }
+
+    &::before {
+      content: '';
+      mask-image: $switch-handle-icon;
+      background-color: transparent;
+      z-index: 1;
+    }
+
+    &::after {
+      border: tokens.get('switch', 'border-width') tokens.get('switch', 'border-style-default')
+        tokens.get('switch', 'handle-enabled-stroke');
     }
 
     &:checked {
-      background-position-x: form-check.$form-switch-width;
+      background-color: tokens.get('switch', 'element-selected-bg');
+      border-color: tokens.get('switch', 'element-selected-stroke');
+
+      &::before,
+      &::after {
+        transform: translateX(
+          calc(tokens.get('switch', 'element-width') - tokens.get('switch', 'element-handle') * 1.5)
+        );
+      }
+
+      &::before {
+        background-color: tokens.get('switch', 'icon-success');
+      }
 
       &::after {
         @include icons-mx.remove-icon;
-        background-color: form-check.$form-switch-color;
-        border-color: form-check.$form-switch-checked-border-color;
-        transform: translateX(2rem);
+        background-color: tokens.get('switch', 'handle-selected');
+        border-color: tokens.get('switch', 'handle-selected-stroke');
+      }
+
+      + .form-check-label {
+        color: tokens.get('switch', 'text-selected-fg');
       }
     }
 
     @include utility-mx.high-contrast-mode {
       transition: none;
-      border: form-check.$form-check-input-border-width solid CanvasText;
+      border: form-check.$form-check-input-border-width tokens.get('switch', 'border-style-default')
+        CanvasText;
 
       &:checked {
         background-color: SelectedItem !important;
       }
 
-      &::after,
-      &:checked::after {
-        max-width: calc(
-          form-check.$form-switch-height - (form-check.$form-check-input-border-width * 2)
-        );
+      &::after {
+        max-width: tokens.get('switch', 'element-handle');
         border-color: Canvas;
         background-color: CanvasText;
       }
+
+      &:checked::after {
+        transform: translateX(
+          calc(
+            tokens.get('switch', 'element-width') - tokens.get('switch', 'element-handle') - tokens.get(
+                'switch',
+                'padding-inline-handle'
+              ) * 2 - form-check.$form-check-input-border-width * 2
+          )
+        );
+      }
     }
 
     &[disabled] {
-      background: form-check.$form-switch-disabled-bg;
+      background-color: tokens.get('switch', 'element-disabled-bg');
+      border: tokens.get('switch', 'border-width') tokens.get('switch', 'border-style-disabled')
+        tokens.get('switch', 'element-disabled-stroke');
+
+      @include color-mx.on-dark-background {
```

Comment:
Light and dark modes are now handled by the tokens through the semantic layer for modes, you shouldn't have to do anything manually.